doc: upgrade path enforcement (#699)
longhorn/website: content/docs/1.5.0/deploy/upgrade/longhorn-manager.md

To avoid crashing existing volumes, as well as switch from the deprecated setting `Guaranteed Engine CPU` to [the new instance manager CPU reservation mechanism](../../../best-practices/#guaranteed-instance-manager-cpu), Longhorn will automatically set `Engine Manager CPU Request` and `Replica Manager CPU Request` from each node based on the deprecated setting value during upgrade. Then, the new global instance manager CPU settings [`Guaranteed Engine Manager CPU`](../../../references/settings/#guaranteed-engine-manager-cpu) and [`Guaranteed Replica Manager CPU`](../../../references/settings/#guaranteed-replica-manager-cpu) won't take effect.
You may need to check the new mechanism and the setting descriptions to see if you need any adjustments.

> ### Upgrading from v1.3.x or older versions

Author: Modified. Thanks.

We only support upgrading to v{{< current-version >}} from v1.4.x. For other versions, please upgrade to v1.4.x first.

> It will fail when upgrading upgrading to v{{< current-version >}} from v1.3.x or older versions.

Author: Modified. Thanks.

#### Upgrade with Kubectl

When you upgrade with kubectl by running this command:

```shell
kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/v{{< current-version >}}/deploy/longhorn.yaml
```

`longhorn-manager` will block the upgrade process and display the reason in logs of the `longhorn-manager` pod.

> User's Longhorn system should remain intact without any impacts except `longhorn-manager`.

Author: Modified. Thanks.

> For recovery, you have to apply the manifest of previous installed version as this command:

Author: Modified. Thanks.

```shell
kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/[previous installed version]/deploy/longhorn.yaml
```

And users might need to delete new components introduced by new version Longhorn manually.

> #### Pre Upgrade

Author: Modified. Thanks.

> To avoid any impacts from upgrading failed by unsupported versions, Longhorn will automatically start a new job to check if the upgrade path is supported when upgrading by `Helm` or as a `Rancher Catalog App`.

Author: Modified. Thanks.

> The `pre-upgrade` job will block the upgrade process and display the reason in logs of the `pre-upgrade` pod.

Reviewer: ```suggestion
The `pre-upgrade` job will block the upgrade process and provide the failiure reason in the logs of the pod.
```

Author: Modified. Thanks.

> User's Longhorn system should remain intact without any impacts.

Reviewer: ```suggestion
During the upgrade failure, the user's Longhorn system should remain intact without any impacts.```

Author: Modified. Thanks.

> For recovery by `Helm`, you can run these commands:

Reviewer: ```suggestion
To recover, you need to run the below commands to rollback to the previously installed revision:
```

Author: Modified. Thanks.

```shell
# get previous installed Longhorn REVISION
helm history longhorn
helm rollback longhorn [REVISION]

# or
helm upgrade longhorn longhorn/longhorn --namespace longhorn-system --version [previous installed version]
```

> For recovery as a `Rancher Catalog App`, you can upgrade the previous installed version at `Rancher App Marketplace` again.

Reviewer: ```suggestion
To recover, you need to upgrade to the previously installed revision at `Rancher App Marketplace` again.
```

Author: Modified. Thanks.
